src/services/SearchService.ts

```diff
@@ -1,7 +1,281 @@
+
+    if (
+      noticeRequirementRule &&
+      noticeRequirementRule.value > intervalSeconds
+    ) {
+      return false;
+    }
+
+    const formattedFirstDay = firstDay.toFormat('ccc').toLowerCase();
+    const lOSRule = (await this.getRule(
+      'length_of_stay',
+      spaceRuleRepository,
+      formattedFirstDay
+    )) as DayOfWeekLOSRule;
+
+    const lOS = dates.length;
+    let minLOS = 0;
+    let maxLOS = 0;
+
+    try {
+      minLOS = lOSRule[formattedFirstDay]?.minLengthOfStay || 0;
+    } catch (e) {
+      //rule not exist
+    }
+
+    try {
+      maxLOS = lOSRule[formattedFirstDay]?.maxLengthOfStay || 0;
+    } catch (e) {
+      //rule not exist
+    }
+
+    return (minLOS === 0 || lOS > minLOS) && (maxLOS === 0 || lOS < maxLOS);
+  }
+
+  private async getRule(
+    ruleName: RulesItemKey,
+    spaceRuleRepository: SpaceRuleRepository,
+    weekDay: null | string = null
+  ) {
+    let rule = await spaceRuleRepository.getRule(ruleName);
+
+    if (!rule || (weekDay && !(weekDay in rule))) {
+      rule = await this.facilityRuleRepository.getRule(ruleName);
+    }
+
+    return rule;
+  }
+
+  private checkSuitableQuantity(space: Space) {
+    const numOfAdults =
+      space.maxNumberOfAdultOccupantsOneof.oneofKind ===
+      'maxNumberOfAdultOccupants'
+        ? space.maxNumberOfAdultOccupantsOneof.maxNumberOfAdultOccupants
+        : 0;
+
+    const numOfChildren =
+      space.maxNumberOfChildOccupantsOneof.oneofKind ===
+      'maxNumberOfChildOccupants'
+        ? space.maxNumberOfChildOccupantsOneof.maxNumberOfChildOccupants
+        : 0;
+
+    const guestCheck = numOfAdults - this.ask.numPaxAdult;
+
+    if (guestCheck < 0) {
+      return false;
+    }
+
+    //if there is a place left from an adult, we give it to a child
+    const childrenCheck =
+      numOfChildren + guestCheck - (this.ask.numPaxChild || 0);
+
+    if (childrenCheck < 0) {
+      return false;
+    }
+
+    //its coefficient for the future so that we can offer multiple rooms for a large group (available for discussion)
+    // const check2 = spaceGuestsCount - guestCount + childrenCount > 0
+    //   ? (spaceGuestsCount + spaceChildrenCount) / (spaceGuestsCount - guestCount + childrenCount)
+    //   : 0;
+
+    return true;
+  }
+
+  private async checkAvailableDates(spaceId, dates: DateTime[]) {
+    const availabilityRepository = new SpaceAvailabilityRepository(
+      this.facilityId,
+      spaceId
+    );
+
+    const spaceStubRepository = new SpaceStubRepository(
+      this.facilityId,
+      spaceId
+    );
+
+    const defaultAvailable = await availabilityRepository.getSpaceAvailability(
+      'default'
+    );
+
+    for (const date of dates) {
+      try {
+        const formattedDate = date.toFormat('yyyy-MM-dd') as FormattedDate;
+
+        let available = await availabilityRepository.getSpaceAvailability(
+          formattedDate
+        );
+
+        if (available.numSpaces === 0) {
```

Comment:
How about the case where intentionally the `numSpaces` is set to `0`, ie. an accommodation provider rams the `numSpaces available to 0 in order to "block" a date. If they do this, `numSpaces === 0` and then it will be assumed that the spaces available are actually the `default` number of spaces.

I think that this would require some editing of the repository, that if the daily override isn't found, instead of return a `numSpaces = 0` object, it should return undefined and this should be dealt with in the logic here (ie. if `!available.numSpaces` then `available = defaultAvailable` (as a `numSpaces = 0` is a legitimate value as well).

Please add a test case for this (ie. `numSpaces = 0` being set as a daily override to simulate "blocking" a space).